bridges/otelzap/core.go

```diff
@@ -0,0 +1,135 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package otelzap provides a bridge between the [go.uber.org/zap] and
+// OpenTelemetry logging.
+package otelzap // import "go.opentelemetry.io/contrib/bridges/otelzap"
+
+import (
+	"context"
+	"slices"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"go.opentelemetry.io/otel/log"
+)
+
+const (
+	bridgeName = "go.opentelemetry.io/contrib/bridge/zapcore"
+)
+
+// Core is a [zapcore.Core] that sends logging records to OpenTelemetry.
+type Core struct {
+	logger log.Logger
+	attr   []log.KeyValue
+	ctx    context.Context
+}
+
+// Compile-time check *Core implements zapcore.Core.
+var _ zapcore.Core = (*Core)(nil)
+
+// NewOTelZapCore creates a new [zapcore.Core] that can be used with zap.New()
+// this instance will translate zap logs to opentelemetry logs and export them.
+func NewOTelZapCore(opts ...Option) zapcore.Core {
```

Comment:
Kind of nit, but as in my previous comment, having `Otel` and `Zap` here is redundant, since that's all there will be in the package.

```suggestion
func NewCore(opts ...Option) zapcore.Core {
```

With this, the call becomes `otelzap.NewCore`, which indicates everything already.